File: ESTRUTURAS/3176/main.rb
# frozen_string_literal:true

def print_formatted(dwarves, n)
  teams = n / 3
  teams.times do |team|
    puts "Time #{team + 1}"
    3.times do |shift|
      puts dwarves[team + teams * shift].join(' ')
    end
    puts
  end
end

dwarves = []
n = gets.to_i
n.times do
  dwarves.push(gets.split)
end
dwarves.sort!.sort! { |a, b| b[1] <=> a[1] }
print_formatted(dwarves, n)

File: ESTRUTURAS/3176/main.dart
import 'dart:io';

void printFormatted(List<List<String>> dwarves, int n) {
  int teams = n ~/ 3;
  for (int team = 0; team < teams; team++) {
    print('Time ${team + 1}');
    for (int shift = 0; shift < 3; shift++) print(dwarves[team + teams * shift].join(' '));
    print('');
  }
}

void main() {
  List<List<String>> dwarves = [];
  int n = int.parse(stdin.readLineSync()!);
  for (int times = 0; times < n; times++) dwarves.add(stdin.readLineSync()!.split(' '));
  dwarves.sort((a, b) => a[0].compareTo(b[0]));
  dwarves.sort((a, b) => b[1].compareTo(a[1]));
  printFormatted(dwarves, n);
}
